### PR TITLE
8171381: [TEST_BUG] [macos] javax/swing/JPopupMenu/7156657/bug7156657.java fails on OS X

### PR DESCRIPTION
**Pull request by azuev-java**
Replaces usage of SunToolkit.realSync() with Robot.waitForIdle()

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8171381](https://bugs.openjdk.java.net/browse/JDK-8171381): [TEST_BUG] [macos] javax/swing/JPopupMenu/7156657/bug7156657.java fails on OS X

### Reviewers
 * [Sergey Bylokhov](https://openjdk.java.net/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3606/head:pull/3606` \
`$ git checkout pull/3606`

Update a local copy of the PR: \
`$ git checkout pull/3606` \
`$ git pull https://git.openjdk.java.net/jdk pull/3606/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3606`

View PR using the GUI difftool: \
`$ git pr show -t 3606`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3606.diff">https://git.openjdk.java.net/jdk/pull/3606.diff</a>

</details>

### CODE REVIEW

**Comment by mrserb** [comment]
The Robot.waitForIdle() is implemented via SunToolkit.realSync(), I am not sure how it might solve the problem.

**Comment by azuev-java** [comment]
> The Robot.waitForIdle() is implemented via SunToolkit.realSync(), I am not sure how it might solve the problem.

It will not solve the problem as there is no problem as such. Test failed twice in two years on the same machine with the problematic OS configuration. Test has not failed before for 6 years and since configuration was fixed has not resurfaced. I'm just using this opportunity to get rid of the non-public API usage in a public test.

**Comment by mrserb** [comment]
> It will not solve the problem as there is no problem as such.

Ok, then please add this thing to the JBS.

File: test/jdk/javax/swing/JPopupMenu/7156657/bug7156657.java [hunk 1]
```diff
@@ -38,8 +37,6 @@
 import javax.swing.JPopupMenu;
 import javax.swing.SwingUtilities;
 
-import sun.awt.SunToolkit;
-
```
Comment (mrserb) [inline test/jdk/javax/swing/JPopupMenu/7156657/bug7156657.java:42]: The "@modules java.desktop/sun.awt" should be removed as well?

Comment (azuev-java) [reply test/jdk/javax/swing/JPopupMenu/7156657/bug7156657.java:42]: Yes, sorry, forgot about it. Fixed.

**Review by mrserb — APPROVED** [review]

**Pull request closed without merging**